uspace/lib/c/generic/rtld/module.c

```diff
@@ -180,13 +180,13 @@ module_t *module_load(rtld_t *rtld, const char *name, mlflags_t flags)
 {
 	elf_finfo_t info;
 	char name_buf[NAME_BUF_SIZE];
-	module_t *m;
+	module_t *m = NULL;
 	int rc;
 
 	m = calloc(1, sizeof(module_t));
 	if (m == NULL) {
 		printf("malloc failed\n");
-		exit(1);
+		goto error;
```

Comment:
What is the failure mode now? I am thinking whether the exit was not there on purpose because a failure in this code is fatal for the task which is being loaded. So do you get a nicer failure now?